File: src/C++/01_sensors.cpp
/*
 * Compiling:
 * g++ -L../../SenseHat/build -lsense-hat 01_sensors.cpp       \
 *   ../../scripts/communication_helpers/CommunicationHelper.h \
 *   ../../scripts/communication_helpers/CommunicationHelper.c \
 *   -lwiringPi
 */

#include "../../SenseHat/src/sense-hat.h"
#include "../../scripts/communication_helpers/CommunicationHelper.h"

int main()
{
  SenseHAT senseHat;

  for(int i = 0; i < 10; i++) {
    SendCommand(START, "temperature_from_pressure_cpp");
    double temperatureFromPressure = senseHat.get_temperature_from_pressure();
    SendCommand(STOP, "temperature_from_pressure_cpp");

    SendCommand(START, "temperature_from_humidity_cpp");
    double temperatureFromHumidity = senseHat.get_temperature_from_humidity();
    SendCommand(STOP, "temperature_from_humidity_cpp");

    SendCommand(START, "pressure_cpp");
    double pressure = senseHat.get_pressure();
    SendCommand(STOP, "pressure_cpp");

    SendCommand(START, "humidity_cpp");
    double humidity = senseHat.get_humidity();
    SendCommand(STOP, "humidity_cpp");
  }

  return 0;
}

File: src/Python/01_sensors.py
#!/usr/bin/env python3

from scripts.communication_helpers.communication_helper import Command, send_message

from sense_hat import SenseHat
sense = SenseHat()

for _ in range(10):
  send_message(Command.START, 'temperature_from_pressure')
  temperature_from_pressure = sense.get_temperature_from_pressure()
  send_message(Command.STOP, 'temperature_from_pressure')

  send_message(Command.START, 'temperature_from_humidity')
  temperature_from_humidity = sense.get_temperature_from_humidity()
  send_message(Command.STOP, 'temperature_from_humidity')

  send_message(Command.START, 'pressure')
  pressure = sense.get_pressure()
  send_message(Command.STOP, 'pressure')

  send_message(Command.START, 'humidity')
  humidity = sense.get_humidity()
  send_message(Command.STOP, 'humidity')
